ggshield/scan/docker.py

```diff
@@ -336,13 +348,15 @@ def docker_scan_archive(
             if file_count == 0:
                 continue
             print()
-            display_heading(f"Scanning layer {info.get_id()}")
-            with RichSecretScannerUI(file_count) as ui:
-                results.extend(
-                    scanner.scan(
-                        layer.files,
-                        scanner_ui=ui,
-                    )
-                )
+            layer_id = info.get_id()
+            if layer_id in layer_id_cache:
+                display_heading(f"Skipping layer {layer_id}: already scanned")
+            else:
+                display_heading(f"Scanning layer {info.get_id()}")
+                with RichSecretScannerUI(file_count) as ui:
+                    layer_results = scanner.scan(layer.files, scanner_ui=ui)
+                if not layer_results.has_policy_breaks:
```

Comment:
A slightly tricky question. We could decide to cache layers that have no secret policy breaks only, instead of no policy breaks at all. This means caching more layers.  
On the other hand, ScanResult does not provide such a property, and it means implementing it by iterating over policy breaks to see if the ScanResult has secrets policy breaks. What do you think ?

Reply:
Assuming we did that: if we ever re-enable again checks for other policy breaks (which I hope we do at some point), then we would have to revisit this because we would potentially silently skip those policy breaks.

That's why I think it's safer to be strict on the conditions required to bypass the scan.

Reply:
Alright, makes sense.